src/dotnet/commands/dotnet-build/CompileContext.cs

```diff
@@ -91,7 +91,7 @@ private bool CompileDendencies(bool incremental)
 
         private bool DependencyNeedsRebuilding(ProjectDescription dependency)
         {
-            var dependencyProjectContext = ProjectContext.Create(dependency.Path, dependency.Framework, new[] { _rootProject.RuntimeIdentifier });
+            var dependencyProjectContext = ProjectContextCache.Default.Create(dependency.Path, dependency.Framework, new[] { _rootProject.RuntimeIdentifier });
```

Comment:
Could you have ProjectContext.Create lookup the cache? Or would that break some clients?